apps/backend/src/handler/authHandler.ts

```diff
@@ -29,10 +30,11 @@ const signupSchema = object({
 	lastname: optional(pipe(string(), trim(), minLength(1))),
 });
 
-const getSession = auth.get("/session", (c) => {
+const getSession = auth.get("/session", async (c) => {
 	const session = c.get("session");
-	if (session) {
-		return c.redirect("/");
+	if (session?.userId) {
+		const user = await getUserById(Number(session.userId));
+		return c.json(user, 200);
```

Comment:
Maybe we should add a meaningful error message to this? Should be considered for every endpoint